### ISSUE TITLE
[Callbacks] Early stopping and model checkpoints

### ISSUE THREAD

**Issue by karnawhat** [ayushkarnawat/profit#59]
Should we follow the conventions used by keras when training models. Instead of passing in the validation loss and the model into the `EarlyStopping().on_epoch_end()` at the end of each training loop, it would be easier (and make more sense) to simply update the logs for the parameters we are trying to monitor.
